torch/utils/_config_module.py

```diff
@@ -157,7 +161,13 @@ def __delattr__(self, name: str) -> None:
         self._is_dirty = True
         # must support delete because unittest.mock.patch deletes
         # then recreate things
-        del self._config[name]
+        self._config[name].user_override = None
```

Comment:
OK, you gotta be careful about this. Because I can explicitly say `config.key = None` and if the default was NOT none then this is not equivalent to "del". It would be much better to have a distinguished singleton like `notset = object()` which we use to do the tests here.

Reply:
I think in many cases this potentially may not matter, because a lot of times we allowed for None in a config key, it was "defer to some default behavior" which we are going to put into the config module to handle for us. But I would rather not poke this bear.

Reply:
Updated tests to show this behaviour, and used a sentinel here.